app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    @functools.wraps(func)
+    def inner(*args, **kwargs) -> Any:
```

Comment:
What if decorator is applied to two different functions that take the same arguments?